feat: use delete process instance C8 API endpoint

konstantinos-mylosis opened the pull request:
## Description

- Introduce new /v2 process instance deletion endpoint
- Introduce process instance deletion request and schema
- Introduce mutation for process instance deletion
- Replace old process instance deletion call with /v2 one using mutation

### Currently
- `/v2/process-instances/${processInstanceKey}/deletion` is used in:
  - instance details view
  - instances list view
- Delete operation is visible in UI -- loading
- Actually deletion occurs asynchronously
- A get process instance occurs with retries so we verify process instance deleted
- Eventually process instance is deleted and is not visible in UI
- Batch Operation seems to be loading forever

### Note
- Batch Operation seems to be loading forever
  - It is because of #41138 not implemented

## Related issues

closes #33014 
closes #33063

konstantinos-mylosis committed:
feat: enhance camunda-api-zod-schemas

add process instance deletion endpoint, request schema
bump up camunda-api-zod-schemas
resolve exposed endpoint

konstantinos-mylosis committed:
feat: introduce process instance deletion request

konstantinos-mylosis committed:
feat: use mutation for process instance deletion

konstantinos-mylosis committed:
test: use mutation for process instance deletion

konstantinos-mylosis committed:
feat: replace old process instance deletion request with new v2 one

konstantinos-mylosis committed:
feat: update camunda-api-zod-schemas version to be published

remcowesterhoud reviewed (commented):
@konstantinos-mylosis Just a few question, probably as I'm not so familiar with the FE. Definitely ask a FE engineer for a proper review 😄. Can't wait to see it in action 🚀 

❓ I don't see anything related to deletion of multiple process instances at once, but this issue closes #33063. Is that right?

konstantinos-mylosis committed:
feat: use new camunda-api-zod-schemas 0.0.28

konstantinos-mylosis committed:
test: update audit log category enum

konstantinos-mylosis committed:
test: fix front end UT after using useDeleteProcessInstance mutation

konstantinos-mylosis commented on client-components/packages/camunda-api-zod-schemas/lib/8.9/process-instance.ts:
```diff
@@ -0,0 +1,34 @@
+	API_VERSION,
+	type Endpoint,
+} from '../common';
+import {type ProcessInstance} from '../8.8';
```
I’m not aware of the best practices here. We have a similar use case [here](https://github.com/camunda/camunda/blob/dbfc3c4e48a318184df98f43b9779be48d34e854/client-components/packages/camunda-api-zod-schemas/lib/8.9/audit-log.ts#L19), and moving the type to 8.9 would require republishing the schema, which may not be worth it.

konstantinos-mylosis committed:
feat: fix useDeleteProcessInstance

pick UseMutationOptions, not partial
use common handleOperationSuccess
on deletion success redirect to process
add a notification that instance deleted

vsgoulart commented on operate/client/package.json:
@konstantinos-mylosis next time please release this on a separate PR. This PR has been open for a couple of weeks already. If anyone else needed to change the defs they might get blocked by you

vsgoulart reviewed (changes requested):
I have meeting now. I'll finish the review later

konstantinos-mylosis committed:
test: remove nested describe

konstantinos-mylosis committed:
feat: keep old mechanism on process instance deletion

notify that Instance is scheduled for deletion
do not wait until process instance is deleted
redirect only when in process instance details view

konstantinos-mylosis committed:
test: update notification on deletion

vsgoulart commented on operate/client/src/modules/utils/processInstance/handleOperationSuccess.ts:
```diff
@@ -0,0 +1,69 @@
+        onInvalidateQueries();
+      } else {
+        queryClient.invalidateQueries({
+          queryKey: ['processInstances'],
```
It's better to add this to `src/modules/queries/queryKeys.ts`. This way it won't get out of sync with the naming

vsgoulart reviewed (approved):
I'm approving it but I believe we should address the 2 comments I made before merging

I did some basic testing and things seems to be working fine

konstantinos-mylosis committed:
feat: move queryKey to keep in sync

removed unnecessary useCallback

konstantinos-mylosis commented on operate/client/src/modules/utils/processInstance/handleOperationSuccess.ts:
```diff
@@ -0,0 +1,69 @@
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+
+  return useCallback(
```
I see `useCallback` is not necessary here

vsgoulart commented:
@konstantinos-mylosis can we merge this?
